### PR TITLE
Humanize error messages

### PR DESCRIPTION
**Pull request by plaindocs**
Pushing this up a place to start the conversation.

### CODE REVIEW

**Comment by plaindocs** [comment]
After settling on wording, need to update the spec definition over in https://github.com/travis-ci/travis-yml/blob/master/spec/travis/yaml_spec.rb to make the tests pass. (Thanks @carlad )

File: lib/travis/yaml.rb [hunk 1]
```diff
@@ -22,32 +22,32 @@ module Yaml
...
+      flagged:         'please email support@travis-ci.org to enable %{given}',
+      irrelevant:      'you used %{key}, but it is not relevant for the %{on_key} %{on_value}',
       unsupported:     '%{key} (%{value}) is not supported on the %{on_key} %{on_value}',
-      required:        'missing required key %{key}',
+      required:        'you need to specify %{key}',
       empty:           'dropping empty section %{key}',
       find_key:        'key %{original} is not known, but %{key} is, using %{key}',
       find_value:      'value %{original} is not known, but %{value} is, using %{value}',
-      clean_key:       'key %{original} contains special characters, using %{key}',
+      clean_key:       'key %{original} contains unsupported characters, using %{key}',
```
Comment (plaindocs) [inline lib/travis/yaml.rb]: Can we be specific about encoding here?

File: lib/travis/yaml.rb [hunk 2]
```diff
@@ -22,32 +22,32 @@ module Yaml
...
       cast:            'casting value %{given_value} (%{given_type}) to %{value} (%{type})',
-      default:         'missing %{key}, defaulting to: %{default}',
-      deprecated:      '%{given} is deprecated',
-      downcase:        'downcasing %{value}',
-      edge:            '%{given} is experimental and might be removed in the future',
-      flagged:         'your repository must be feature flagged for %{given} to be used',
-      irrelevant:      'specified %{key}, but this setting is not relevant for the %{on_key} %{on_value}',
+      default:         'missing %{key}, using the default %{default}',
+      deprecated:      '%{given} is deprecated', # Do we need to say stop using it? or?
+      downcase:        'lowercasing %{value}',
```
Comment (plaindocs) [inline lib/travis/yaml.rb]: maybe need to reword to this to `using lower case of` or somthing to be more consistent

File: lib/travis/yaml.rb [hunk 3]
```diff
@@ -22,32 +22,32 @@ module Yaml
...
+      edge:            '%{given} is experimental and might be removed without notice', #is without notice accurate?
+      flagged:         'please email support@travis-ci.org to enable %{given}',
```
Comment (plaindocs) [inline lib/travis/yaml.rb]: .org / .com / eterprise. But something slightly helpful

File: lib/travis/yaml.rb [hunk 4]
```diff
@@ -22,32 +22,32 @@ module Yaml
     # these are meant as examples. we might want to determine the translation
     # in the client
     MSGS = {
-      alert:           'using a plain string as a secure',
+      alert:           'this string should probably be encrypted',
```
Comment (plaindocs) [inline lib/travis/yaml.rb]: How do we tell them which string? I'm guessing we don't have a variable due to potential length of the string

**Comment by plaindocs** [comment]
What is the difference between alias and deprecated, when to deprecate you alias. :-p 

Al la `if` to `on` kinda stuff over on https://github.com/travis-ci/docs-travis-ci-com/pull/1400

**Comment by joecorcoran** [comment]
As I understand it, `alias` can also be used as an `info`-level message, just to explain why something was changed. Meaning, we don't only alias things when one is deprecated. `matrix` and `jobs` are both the same and will be supported long term, I think.

**Comment by plaindocs** [comment]
Thanks for fixing the tests Joe, I was leaving in case we to-and-froed a little more. I'll take another look at this soonish.

File: lib/travis/yaml.rb (hunk 3 above)
Comment (carlad) [reply lib/travis/yaml.rb]: I've added a `host:` key to config but am wondering if we can simply default to `.com` seeing as how all support emails end up in the same place? (not sure how this affects enterprise)

**Comment by carlad** [comment]
I've updated with most of Sam's suggestions and merged this for now. We'll need to keep iterating on the messages in any case so figured it was best to move this forward.

**Comment by plaindocs** [comment]
Thanks Carla! I can take another look in a week or so if there are any further changes?

**Pull request merged**